### PR TITLE
[4165] Add 'else' clause to call method

### PR DESCRIPTION
**Pull request by avinhurry**
## Context

There have been cases where candidates haven't recieved their rejection email. This is because the `call` is missing an `else` clause. 

## Changes proposed in this pull request

Make the `all_applications_rejected` email the default to account for edge cases where candidates were not receiving rejection emails.

## Guidance to review

- Are there any scenarios that have been missed in the tests.
- Do we need to add logic which sends a different default email based on whether it's before or after the end of cycle.

### CODE REVIEW

**Comment by avinhurry** [comment]
[Candidates don't always receive rejection emails.](https://trello.com/c/EJHj3k5m/4165-candidates-dont-always-receive-rejection-emails)

**Review by stevehook — COMMENTED** [review]
Looks good. I think not all of the test cases are needed because they test combinations of statuses that shouldn't be possible in the wild.

**Comment by avinhurry** [comment]
> Looks good. I think not all of the test cases are needed because they test combinations of statuses that shouldn't be possible in the wild.

Thanks @stevehook! Need to get my head around the business logic.

File: spec/services/send_candidate_rejection_email_spec.rb [hunk 1]
```diff
@@ -74,6 +79,76 @@
...
+          allow(CandidateMailer).to receive(:application_rejected_all_applications_rejected).and_return(mail)
+          described_class.new(application_choice: application_choice).call
+        end
+
+        it 'sends the all_applications_rejected email to the candidate' do
+          expect(CandidateMailer).to have_received(:application_rejected_all_applications_rejected).with(application_choice)
+        end
+      end
+
+      describe 'applications have had offers deferred and another is declined' do
```
Comment (stevehook) [inline spec/services/send_candidate_rejection_email_spec.rb:139]:
```suggestion
      describe 'applications have had offers deferred and another is rejected' do
```

File: spec/services/send_candidate_rejection_email_spec.rb [hunk 2]
```diff
@@ -74,6 +79,76 @@
...
+          allow(CandidateMailer).to receive(:application_rejected_all_applications_rejected).and_return(mail)
+          described_class.new(application_choice: application_choice).call
+        end
+
+        it 'sends the all_applications_rejected email to the candidate' do
+          expect(CandidateMailer).to have_received(:application_rejected_all_applications_rejected).with(application_choice)
+        end
+      end
+
+      describe 'when applications are declined and another is declined' do
```
Comment (stevehook) [inline spec/services/send_candidate_rejection_email_spec.rb:111]:
```suggestion
      describe 'when applications are declined and another is rejected' do
```

File: spec/services/send_candidate_rejection_email_spec.rb [hunk 3]
```diff
@@ -74,6 +79,76 @@
...
+          allow(CandidateMailer).to receive(:application_rejected_all_applications_rejected).and_return(mail)
+          described_class.new(application_choice: application_choice).call
+        end
+
+        it 'sends the all_applications_rejected email to the candidate' do
+          expect(CandidateMailer).to have_received(:application_rejected_all_applications_rejected).with(application_choice)
+        end
+      end
+
+      describe 'applications have had offers withdrawn and another is declined' do
```
Comment (stevehook) [inline spec/services/send_candidate_rejection_email_spec.rb:125]:
```suggestion
      describe 'applications have had offers withdrawn and another is rejected' do
```

File: spec/services/send_candidate_rejection_email_spec.rb [hunk 4]
```diff
@@ -74,6 +79,76 @@
...
+          allow(CandidateMailer).to receive(:application_rejected_all_applications_rejected).and_return(mail)
+          described_class.new(application_choice: application_choice).call
+        end
+
+        it 'sends the all_applications_rejected email to the candidate' do
+          expect(CandidateMailer).to have_received(:application_rejected_all_applications_rejected).with(application_choice)
+        end
+      end
+
+      describe 'when applications are not sent and another is rejected' do
```
Comment (stevehook) [inline spec/services/send_candidate_rejection_email_spec.rb:97]: I think you can delete this case too.

**Review by stevehook — APPROVED** [review]
Looks good. Let's merge. Some very minor nits inline.
